oxcache/src/cache/mod.rs

```diff
@@ -13,12 +14,14 @@ pub mod bucket;
 #[derive(Debug)]
 pub struct Cache {
     buckets: RwLock<HashMap<Chunk, Arc<RwLock<ChunkState>>>>,
+    zone_to_entry: Mutex<Vec<Vec<Chunk>>>
 }
 
 impl Cache {
-    pub fn new() -> Self {
+    pub fn new(num_zones: usize) -> Self {
         Self {
             buckets: RwLock::new(HashMap::new()),
+            zone_to_entry: Mutex::new(vec![Vec::new(); num_zones])
```

Comment:
I think we should specify with capacity matching chunks per zone here to avoid reallocations.

```suggestion
            zone_to_entry: Mutex::new(vec![Vec::with_capacity(chunks_per_zone); num_zones])
```